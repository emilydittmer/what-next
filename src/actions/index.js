export const setSearchShow = show => ({
  type: 'SET_SEARCH_SHOW',
  show
})