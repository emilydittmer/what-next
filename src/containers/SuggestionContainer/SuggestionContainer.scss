.suggestion-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-column-gap: 10px;
}

h2 {
  margin-left: 20px;
}