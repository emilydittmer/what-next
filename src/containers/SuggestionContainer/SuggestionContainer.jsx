import React, { Component } from 'react';
import './SuggestionContainer.scss';
import { fetchSuggestedShows } from '../../utils/apiCalls'
import { connect } from 'react-redux';
import { grabSuggestedShows } from '../../actions';
import SuggestionCard from '../SuggestionCard/SuggestionCard'

class SuggestionContainer extends Component{
  constructor(props){
    super(props);
    this.state = {
      error: ''
    }
  }

  componentDidMount() {
    fetchSuggestedShows(this.props.search.id)
      .then(shows => this.props.grabSuggestedShows(shows))
      .catch(this.setState({ error: "Error fetching suggestions" }));
  }
  showValue() {
    if(this.props.suggestion) {
      return(<h2>{this.state.error}</h2>)
    } else {
      return this.props.suggestions.map(suggestion => (
        <SuggestionCard {...suggestion} key={suggestion.id} />
      ))
    }
  }
      
  render(){
    return(
      <section>
        <h3>Shows like {this.props.search.name}: </h3>
        {this.showValue()}
      </section>
    )
  }
}

const mapStateToProps = state => ({
  search: state.search,
  suggestions: state.suggestions
})

const mapDispatchToProps = dispatch => ({
  grabSuggestedShows: shows => dispatch(grabSuggestedShows(shows))
})

export default connect(mapStateToProps, mapDispatchToProps)(SuggestionContainer);