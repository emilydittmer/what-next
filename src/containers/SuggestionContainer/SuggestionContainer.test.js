import React from 'react';
import { shallow } from 'enzyme';
import { setSearchShow, grabSuggestedShows } from '../../actions'
import { SuggestionContainer, mapStateToProps, mapDispatchToProps } from './SuggestionContainer';

describe('SuggestionContainer', ()=> {
  let wrapper, props;

  beforeEach(() => {
    props = {
      search: {search: {id: 1, name:'The Office'}},
      suggestions: [{id: 1, name:'The Office'}, {id: 1, name:'The Office'}]
    },
    wrapper = shallow(<SuggestionContainer {...props}/>)
  })

  it('should match the snapshot', () => {
    wrapper.setState({error: ''})
    expect(wrapper).toMatchSnapshot()
  })

  it('should render an error', () => {
    wrapper.setState({error: 'Error'})
    expect(wrapper).toMatchSnapshot()
  })

  // describe("mapStateToProps", () => {
  //   it("should return an object with the searched show", () => {

  //   });
  // });

  describe("mapDispatchToProps", () => {
    it("calls dispatch with a grabSuggestedShows action", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = grabSuggestedShows("The Office");
      const mappedDispatch = mapDispatchToProps(mockDispatch);
      mappedDispatch.grabSuggestedShows("The Office");
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
}) 