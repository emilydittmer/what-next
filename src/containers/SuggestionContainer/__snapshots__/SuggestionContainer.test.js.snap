// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionContainer should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`SuggestionContainer should render an error 1`] = `ShallowWrapper {}`;
